node-js-basics-101/node_modules/asynquence-contrib/contrib.src.js

```diff
@@ -0,0 +1,2360 @@
+		return Ar.filter(seq,makeDistinctFilterer(/*keepAll=*/true));
+	};
+
+	Ar.distinctConsecutive = Ar.distinctUntilChanged = function $$react$distinct$consecutive(seq) {
+		return Ar.filter(seq,makeDistinctFilterer(/*keepAll=*/false));
+	};
+
+	Ar.filter = function $$react$filter(seq,filterer){
+		function reactor(next,registerTeardown) {
+			function trigger(){
+				var messages = ASQ.messages.apply(ø,arguments);
+
+				if (filterer && filterer.apply(ø,messages)) {
+					// fire off reactive sequence instance
+					next.apply(ø,messages);
+				}
+
+				// keep sequence going
+				return messages;
+			}
+
+			// sequence-stream event listener
+			def.seq.val(trigger);
+
+			// listen for stop() of reactive sequence
+			registerTeardown(function $$teardown(){
+				def = filterer = null;
+			});
+		}
+
+		// observe sequence-stream
+		var def = tapSequences(seq)[0];
+
+		if (!def) return;
+
+		return Ar(reactor);
+	};
+
+	Ar.fromObservable = function $$react$from$observable(obsv){
+		function reactor(next,registerTeardown){
+			// process buffer (if any)
+			buffer.forEach(next);
+			buffer.length = 0;
+
+			// start non-buffered notifications?
+			if (!buffer.complete) {
+				notify = next;
+			}
+
+			registerTeardown(function $$teardown(){
+				obsv.dispose();
+			});
+		}
+
+		function notify(v) {
+			buffer.push(v);
+		}
+
+		var buffer = [];
+
+		obsv.subscribe(
+			function $$on$next(v){
+				notify(v);
+			},
+			function $$on$error(){},
+			function $$on$complete(){
+				buffer.complete = true;
+				obsv.dispose();
+			}
+		);
+
+		return Ar(reactor);
+	};
+
+	ASQ.extend("toObservable",function $$extend(api,internals){
+		return function $$to$observable(){
+			function init(observer) {
+				function define(pair){
+					function listen(){
+						var args = ASQ.messages.apply(ø,arguments);
+						observer[pair[1]].apply(observer,
+							args.length == 1 ? [args[0]] : args
+						);
+						return args;
+					}
+
+					api[pair[0]](listen);
+				}
+
+				[["val","onNext"],["or","onError"]]
+				.forEach(define);
+			}
+
+			return Rx.Observable.create(init);
+		};
+	});
+
+	function tapSequences() {
+		function tapSequence(seq) {
+			// temporary `trigger` which, if called before being replaced
+			// below, creates replacement proxy sequence with the
+			// event message(s) re-fired
+			function trigger() {
+				var args = ARRAY_SLICE.call(arguments);
+				def.seq = Ar(function $$react(next){
+					next.apply(ø,args);
+				});
+			}
+
+			if (ASQ.isSequence(seq)) {
+				var def = { seq: seq };
+
+				// listen for events from the sequence-stream
+				seq.val(function $$val(){
+					trigger.apply(ø,arguments);
+					return ASQ.messages.apply(ø,arguments);
+				});
+
+				// make a reactive sequence to act as a proxy to the original
+				// sequence
+				def.seq = Ar(function $$react(next){
+					// replace the temporary trigger (created above)
+					// with this proxy's trigger
+					trigger = next;
```

Comment:
__[jshint]__
* Reassignment of 'trigger', which is is a function. Use 'var' or 'let' to declare bindings that may change.

<!-- Comment by [SideCI](https://sideci.com) -->